dags/utils/slack_operator.py

```diff
@@ -1,88 +1,171 @@
-from airflow.hooks.base_hook import BaseHook
-from airflow.contrib.operators.slack_webhook_operator import SlackWebhookOperator
+import os
+import datetime
+from pendulum import timezone
+
+from cron_descriptor import get_description
+from airflow.hooks.base import BaseHook
+from airflow.providers.slack.operators.slack_webhook import SlackWebhookOperator
 
 # This is the Conn Id that we set when creating the connection in the Airflow dashboard
 # in Admin > Connections.
 SLACK_CONN_ID = "slack"
 
+DEPLOYMENT_ENVIRONMENT = os.getenv("ENVIRONMENT", "development")
+
+
+def format_schedule(schedule_interval):
+    """
+    Convert a schedule interval to a human-readable format.
+
+    This function handles different types of schedule intervals:
+    - None: Returns "None" as the description.
+    - str: Assumes the string is a cron expression and tries to convert it to a human-readable format.
+           If conversion fails, it returns the cron expression as is.
+    - datetime.timedelta: Converts the timedelta to a human-readable string, breaking it down into days, hours, minutes, and seconds.
+    - Other types: Converts the interval to a string representation.
+
+    Args:
+        schedule_interval: The schedule interval to format. It can be None, a string (cron expression), or a datetime.timedelta.
+
+    Returns:
+        str: A human-readable description of the schedule interval.
+    """
+    if schedule_interval is None:
+        schedule_description = "None"
+    elif isinstance(schedule_interval, str):
+        try:
+            schedule_description = get_description(schedule_interval)
+        except Exception:
+            schedule_description = f"Cron expression: {schedule_interval}"
+    elif isinstance(schedule_interval, datetime.timedelta):
+        # Format timedelta to human-readable string
+        total_seconds = int(schedule_interval.total_seconds())
+        periods = [
+            ("day", 86400),  # 60 * 60 * 24
+            ("hour", 3600),  # 60 * 60
+            ("minute", 60),
+            ("second", 1),
+        ]
+        parts = []
+        for period_name, period_seconds in periods:
+            if total_seconds >= period_seconds:
+                period_value, total_seconds = divmod(total_seconds, period_seconds)
+                part = f"{period_value} {period_name}{'s' if period_value > 1 else ''}"
+                parts.append(part)
+        schedule_description = "Every " + ", ".join(parts)
+    else:
+        schedule_description = str(schedule_interval)
+    return schedule_description
+
 
 def get_central_time_exec_data(context):
-    from pendulum import timezone
```

Comment:
my memory on this is pretty vague, but I think we stashed the `import` in here because it's good to be mindful of what libs are imported from within the `dags` directory, because Airflow actually runs all of the DAG code every time it checks for new dags. 

probably not a big deal—and we should have added a comment here in the first place!

Reply:
I'm sure you're right - thank you for saying this! When I saw the import down inside the function, I didn't understand the implications of it being there, but this makes sense and is a TIL. I'll move this back down here, 100%.

Reply:
https://github.com/cityofaustin/atd-airflow/pull/261/commits/39ac9a93e5b7564b15378769db1531abdea06146